protocols/bftcosi/bftcosi.go

```diff
@@ -97,23 +102,46 @@ type ProtocolBFTCoSi struct {
 
 	// finale signature that this BFTCoSi round has produced
 	finalSignature *BFTSignature
+
+	// our index in the Roster list
+	index int
+	// prepareSignature is the signature generated during the prepare phase
+	// This signature is adapted according to the exceptions that occured during
+	// the prepare phase.
+	prepareSignature []byte
+	// commitSignature is the signature generated during the commit phase
+	// This signature is adapted according to the exceptions that occured during
+	// the commit phase.
+	commitSignature []byte
```

Comment:
This is for the first round of the protocol where BFTCoSi counts how many nodes are OK to sign off. I also think that the structure could be a little bit more lightweight for what it is. Any suggestions?